esmvalcore/_citation.py

```diff
@@ -0,0 +1,229 @@
+    """
+    product_name = os.path.splitext(filename)[0]
+    info_urls = []
+    json_urls = []
+    product_tags = []
+    for item in provenance.records:
+        reference_attr = item.get_attribute('attribute:' + 'references')
+        # get cmip6 citation info
+        value = item.get_attribute('attribute:' + 'mip_era')
+        if 'CMIP6' in list(value):
+            url_prefix = _make_url_prefix(item.attributes)
+            info_urls.append(_make_info_url(url_prefix))
+            json_urls.append(_make_json_url(url_prefix))
+        if reference_attr:
+            # get recipe citation tags
+            if item.identifier.namespace.prefix == 'recipe':
+                product_tags += list(reference_attr)
+            # get diagnostics citation tags
+            if item.get_attribute('attribute:' + 'script_file'):
+                product_tags += list(reference_attr)
+
+    # get other references information recorded by provenance
+    tags = set(_clean_tags(product_tags + [ESMVALTOOL_PAPER_TAG]))
+    for item in provenance.records:
+        reference_attr = item.get_attribute('attribute:' + 'references')
+        if reference_attr:
+            value = set(_clean_tags(reference_attr))
+            if not value.issubset(tags):
+                info_urls += list(reference_attr)
+
+    _save_citation_info(product_name, product_tags, json_urls, info_urls)
+
+
+def _save_citation_info(product_name, product_tags, json_urls, info_urls):
+    citation_entries = [ESMVALTOOL_PAPER]
+
+    # save CMIP6 url_info, if any
+    # save any refrences info that is not related to recipe or diagnostics
+    if info_urls:
+        with open(f'{product_name}_data_citation_info.txt', 'w') as file:
+            file.write('\n'.join(list(set(info_urls))))
+
+    # convert json_urls to bibtex entries
+    for json_url in json_urls:
+        cmip_citation = _collect_cmip_citation(json_url)
+        if cmip_citation:
+            citation_entries.append(cmip_citation)
+
+    # convert tags to bibtex entries
+    if REFERENCES_PATH and product_tags:
+        # make tags clean and unique
+        tags = list(set(_clean_tags(product_tags)))
+        for tag in tags:
+            citation_entries.append(_collect_bibtex_citation(tag))
+
+    with open(f'{product_name}_citation.bibtex', 'w') as file:
+        file.write('\n'.join(citation_entries))
+
+
+def _clean_tags(tags):
+    """Clean the tags that are recorded as str by provenance."""
+    pattern = re.compile(r'\w+')
+    return pattern.findall(str(tags))
+
+
+def _get_response(url):
+    """Return information from CMIP6 Data Citation service in json format."""
+    json_data = False
+    if url.lower().startswith('https'):
+        try:
+            response = requests.get(url)
+            if response.status_code == 200:
+                json_data = response.json()
+            else:
+                logger.info('Error in the CMIP json link: %s', url)
```

Comment:
```suggestion
                logger.warning('Error in the CMIP6 citation link: %s', url)
```
This should not happen unless the data is badly formatted, so I think this should be a warning, right?